opencl-aot/source/main.cpp

```diff
@@ -414,7 +414,14 @@ int main(int Argc, char *Argv[]) {
   CLProgramUPtr ProgramUPtr(std::move(Progs[0]));
 
   // step 7: set OpenCL build options
-  std::string BuildOptions = OptBuildOptions;
+  std::string BuildOptsStr;
+  if (!OptBuildOptions.empty()) {
+    BuildOptsStr = *OptBuildOptions.begin();
```

Comment:
Can range-based loop be used here? Smth like that:
```
if (!OptBuildOptions.empty()) {
    for (const auto& BuildOption: OptBuildOptions) {
      BuildOptions += BuildOption + ' ';
    }
}
```

Reply:
Do we care about the extra space delimiter at the end of the string? If not, then that's an option

Reply:
Sorry, missed this thread. Yeah, it doesn't matter if extra space is at the beginning or at the end of the string.